ext/zlib/zlib.c

```diff
@@ -808,24 +808,15 @@ static bool zlib_create_dictionary_string(HashTable *options, char **dict, size_
 					zend_string **end, **ptr = strings - 1;
 
 					ZEND_HASH_FOREACH_VAL(dictionary, cur) {
-						*++ptr = zval_get_string(cur);
-						ZEND_ASSERT(*ptr);
-						if (ZSTR_LEN(*ptr) == 0 || EG(exception)) {
+						*++ptr = zval_try_get_string(cur);
+						if (*ptr == NULL || ZSTR_LEN(*ptr) == 0 || zend_str_has_nul_byte(*ptr)) {
```

Comment:
Something else I'm noticing, is that if the `option_buffer` zval is a string we don't check if it doesn't have a nul byte AFAICS.

Reply:
That is correct, because a dictionary string is a char array of zero terminated strings (otherwise you could only pass a single word).